src/category_theory/limits/preserves/shapes/zero.lean

```diff
@@ -0,0 +1,93 @@
+/-
+Copyright (c) 2022 Markus Himmel. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Markus Himmel
+-/
+import category_theory.limits.preserves.shapes.terminal
+import category_theory.limits.shapes.zero
+
+/-!
+# Preservation of zero objects and zero morphisms
+We define the class `preserves_zero_morphisms` and show basic properties.
+## Main results
+We provide the following results:
+* Equivalences preserve zero morphisms;
+* if both categories involved have a zero object, then a functor preserves zero morphisms if and
+  only if it preserves the zero object;
+* functors which preserve initial or terminal objects preserve zero morphisms.
+-/
+
+universes v₁ v₂ u₁ u₂
+
+noncomputable theory
+
+open category_theory
+open category_theory.limits
+
+namespace category_theory.functor
+variables {C : Type u₁} [category.{v₁} C] {D : Type u₂} [category.{v₂} D]
+
+section zero_morphisms
+variables [has_zero_morphisms C] [has_zero_morphisms D]
+
+/-- A functor preserves zero morphisms if it sends zero morphisms to zero morphisms. -/
+class preserves_zero_morphisms (F : C ⥤ D) : Prop :=
+(map_zero' : ∀ (X Y : C), F.map (0 : X ⟶ Y) = 0 . obviously)
+
+@[simp]
+protected lemma map_zero (F : C ⥤ D) [preserves_zero_morphisms F] (X Y : C) :
+  F.map (0 : X ⟶ Y) = 0 :=
+preserves_zero_morphisms.map_zero' _ _
+
+lemma equivalence_preserves_zero_morphisms (F : C ≌ D) : preserves_zero_morphisms F.functor :=
```

Comment:
Is this a special case of something slightly more general, maybe that adjoints preserve zero morphisms? I don't know the maths here enough to be sure

Reply:
Earlier today I proved on paper that fully faithful is enough, and I'll generalize the proof soon. lt might also be true for adjoints, I'll think about that.